.github/workflows/hq-smoke-tests.yml

```diff
@@ -11,19 +11,32 @@ on:
   repository_dispatch:
     types: [ deploy_success ]
   workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to run tests against'
+        required: true
+        default: 'staging'
+        type: choice
+        options:
+          - staging
+          - production
 
 jobs:
   set_matrix:
     runs-on: ubuntu-latest
     outputs:
-      matrix: ${{ steps.set-matrix.outputs.matrix || steps.set-matrix-default.outputs.matrix }}
+      matrix: ${{ steps.set-matrix-deploy.outputs.matrix || steps.set-matrix-manual.outputs.matrix || steps.set-matrix-default.outputs.matrix }}
     steps:
-      - id: set-matrix
-        if: ${{ github.event.client_payload.environment != null }}
+      - id: set-matrix-deploy
+        if: ${{ github.event_name == 'repository_dispatch' }}
         run: |
           echo "::set-output name=matrix::{\"environment\": [\"${{ github.event.client_payload.environment }}\"]}"
+      - id: set-matrix-manual
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        run: |
+          echo "::set-output name=matrix::{\"environment\": [\"${{ inputs.environment }}\"]}"
```

Comment:
Does this mean that on manual trigger the scripts will only execute on Staging as `default: 'staging'` is set in the inputs block?

Reply:
When triggered manually you can select the environment but if you don't it will be 'staging'. Prior introducing the matrix the env would default to staging if it was not set.

I could change it so that if it's not set it will run on both envs.